src/extension.ts

```diff
@@ -164,3 +176,43 @@ function convertLogLevel(commonLogLevel: string) {
             return "FINE";
     }
 }
+
+function collectUserSettings(): string[] {
```

Comment:
I'm curious why not just use the plain object but use the key map here. The key map is error-prone since it doesn't provide the strong type verification and spell check. And it is tedious to encode/decode the key/values  in the client/server side.